packages/aws-sdk-signers/src/aws_sdk_signers/signers.py

```diff
@@ -789,6 +800,113 @@ async def _compute_payload_hash(
         return checksum.hexdigest()
 
 
+class AsyncEventSigner:
+    def __init__(
+        self,
+        *,
+        signing_properties: SigV4SigningProperties,
+        identity: AWSCredentialIdentity,
+        initial_signature: bytes,
+    ):
+        self._signing_properties = signing_properties
+        self._identity = identity
+        self._prior_signature = initial_signature
+        self._signing_lock = asyncio.Lock()
+
+    async def sign_event(
+        self,
+        *,
+        event_message: "EventMessage",
+        event_encoder_cls: type["EventHeaderEncoder"],
+    ) -> "EventMessage":
+        async with self._signing_lock:
+            # Copy and prepopulate any missing values in the
+            # signing properties.
+            new_signing_properties = SigV4SigningProperties(  # type: ignore
+                **self._signing_properties
+            )
+            # TODO: If date is in properties, parse a datetime from it.
+            date_obj = datetime.datetime.now(datetime.UTC)
+            if "date" not in new_signing_properties:
+                new_signing_properties["date"] = date_obj.strftime(
+                    SIGV4_TIMESTAMP_FORMAT
+                )
+
+            timestamp = new_signing_properties["date"]
+            headers: dict[str, str | bytes | datetime.datetime] = {":date": date_obj}
+            encoder = event_encoder_cls()
+            encoder.encode_headers(headers)
+            encoded_headers = encoder.get_result()
+
+            payload = event_message.encode()
```

Comment:
This doesn't seem right. You're signing the entire encoded event and then modifying it. Shouldn't this just be the payload?

Reply:
oh my god it is actually like this

Reply:
This is correct, but weird.  When sigv4 signing an event, you encode the entire unsigned message as the payload to sign, and then the signed message will have only two headers: `:date` and `:chunk-signature`.